### PR TITLE
Remove unnecessary I prefix interfaces

### PR DESCRIPTION
**Pull request by dapplion**
**Motivation**

- From https://github.com/ChainSafe/lodestar/issues/3652 remove all legacy cases of unnecessary I prefixed types that are interfaces

**Description**

Almost all changes are performed automatically with script below. Non script changes:
- Remove 2 unused interfaces
- Rename duplicate type `Logger` imported from winston

```bash
#!/bin/bash

# Removes all unnecessary uses of `interface IType`, replacing them for `type Type`
# Skips instances with there's some class that already exists with name Type
# Note: this script takes multiple minutes to execute, and modifies hundreds of files

LODESTAR_DIRECTORY="/home/lion/Code/eth2.0/lodestar"
SRC_DIRECTORY="$LODESTAR_DIRECTORY/packages"

items=$(find "$SRC_DIRECTORY" -name "*.ts" -type f -exec grep -H "interface I" {} \; \
  | sed -E 's/^.+interface[[:space:]]+([[:alnum:]]+).+$/\1/g' \
  | sort -u)

echo Found matches: $items

for item in $items
do
  item_no_prefix=${item:1}  # remove first character
  
  if grep -rl "class $item_no_prefix" --include \*.ts $SRC_DIRECTORY >/dev/null; then
    echo "File(s) containing 'class $item_no_prefix' found, skipping changes for $item"
  else
    echo "replacing $item"
    find "$SRC_DIRECTORY" -name "*.ts" -type f -exec sed -i "s/interface[[:space:]]\+$item[[:space:]]*{/type $item_no_prefix = {/g" {} \;
    find "$SRC_DIRECTORY" -name "*.ts" -type f -exec sed -i "s/$item/$item_no_prefix/g" {} \;
  fi
done

# Format with prettier to add ; to type declarations

$LODESTAR_DIRECTORY/node_modules/.bin/prettier "$SRC_DIRECTORY/**/*.ts" --write
```

The script output is 
```

Found matches IApiArgs IApiOptions IAttesterStatus IAttnetsService IBatchDepositEvents IBeaconChain IBeaconChainLc IBeaconClock IBeaconDb IBeaconExtraArgs IBeaconNodeInitModules IBeaconNodeModules IBeaconNodeOptions IBeaconParamsArgs IBeaconPaths IBeaconStateTransitionMetrics IBeaconSync IBlockFilterOptions IBlockProposerData IBlsVerifier ICachedGenesis IChainArgs ICliCommand IClock ICounter IDatabaseApiOptions IDatabaseController IDatabaseOptions IDiscv5Events IDistanceEntry IDownloadGenericTestsOptions IDownloadTestsOptions IENRArgs IEpochShuffling IEth1Args IEth1DataWithTimestamp IEth1ForBlockProduction IEth1Provider IEth1Streamer IEthJsonRpcReturnTypes IExecutionBuilder IExecutionEngine IExportArgs IFilterOptions IForkChoice IForkChoiceStore IForkConfig IForkDigestContext IForkInfo IGauge IGenesisBuilder IGenesisBuilderKwargs IGenesisResult IGlobalPaths IGlobalSingleArgs IGossipEvents IGossipModules IGossipTopic IGossipTopicCache IHistogram IHttpClient IImportArgs IIndexAttestationData IInterchangeCompleteV4 IInterchangeLodestar IInterchangeV5 IJsonRpcBackend IJsonRpcHttpClient IKeyValue ILevelDbControllerMetrics ILevelDBOptions ILibp2pOptions ILightClientStore ILogArgs ILogger ILoggerOptions IMarkdownSection IMetadataModules IMetadataOpts IMetricsArgs IMinMaxSurround IMockChainParams InboundRateLimitQuota INetwork INetworkArgs INetworkInitModules INetworkModules INetworkOptions InstanceCodeOptions IPeerRpcScoreStore IPeerRpcScoreStoreModules IPeerSummary IPersistedKeysBackend IProof IQueueMetrics IReadonlyEpochShuffling IReqRespBeaconNode IRequestErrorMetadata IRewardPenaltyItem IRpcPayload IRpcResponse IRpcResponseError ISignedBLStoExecutionChange ISignedVoluntaryExitData ISimpleCase ISimpleStruct ISlashingProtection ISlotRange ISpecTestOptions IStateRegenerator ISubnetsService ISyncArgs ISyncCommitteeJob ISyncModule ISyncModules ITestCase IUncompressState IUser IValidatorEvents IValidatorGeneratorOpts IValidatorMonitor IWinstonInfoArg
replacing IApiArgs
replacing IApiOptions
replacing IAttesterStatus
File(s) containing 'class AttnetsService' found, skipping changes for IAttnetsService
replacing IBatchDepositEvents
File(s) containing 'class BeaconChain' found, skipping changes for IBeaconChain
File(s) containing 'class BeaconChainLc' found, skipping changes for IBeaconChainLc
replacing IBeaconClock
File(s) containing 'class BeaconDb' found, skipping changes for IBeaconDb
replacing IBeaconExtraArgs
replacing IBeaconNodeInitModules
replacing IBeaconNodeModules
File(s) containing 'class BeaconNodeOptions' found, skipping changes for IBeaconNodeOptions
replacing IBeaconParamsArgs
replacing IBeaconPaths
replacing IBeaconStateTransitionMetrics
File(s) containing 'class BeaconSync' found, skipping changes for IBeaconSync
replacing IBlockFilterOptions
replacing IBlockProposerData
File(s) containing 'class BlsVerifier' found, skipping changes for IBlsVerifier
replacing ICachedGenesis
replacing IChainArgs
replacing ICliCommand
File(s) containing 'class Clock' found, skipping changes for IClock
replacing ICounter
replacing IDatabaseApiOptions
replacing IDatabaseController
replacing IDatabaseOptions
replacing IDiscv5Events
replacing IDistanceEntry
replacing IDownloadGenericTestsOptions
replacing IDownloadTestsOptions
replacing IENRArgs
replacing IEpochShuffling
replacing IEth1Args
replacing IEth1DataWithTimestamp
File(s) containing 'class Eth1ForBlockProduction' found, skipping changes for IEth1ForBlockProduction
File(s) containing 'class Eth1Provider' found, skipping changes for IEth1Provider
replacing IEth1Streamer
replacing IEthJsonRpcReturnTypes
File(s) containing 'class ExecutionBuilder' found, skipping changes for IExecutionBuilder
File(s) containing 'class ExecutionEngine' found, skipping changes for IExecutionEngine
replacing IExportArgs
replacing IFilterOptions
File(s) containing 'class ForkChoice' found, skipping changes for IForkChoice
File(s) containing 'class ForkChoiceStore' found, skipping changes for IForkChoiceStore
replacing IForkConfig
replacing IForkDigestContext
replacing IForkInfo
File(s) containing 'class Gauge' found, skipping changes for IGauge
File(s) containing 'class GenesisBuilder' found, skipping changes for IGenesisBuilder
replacing IGenesisBuilderKwargs
replacing IGenesisResult
replacing IGlobalPaths
replacing IGlobalSingleArgs
replacing IGossipEvents
replacing IGossipModules
File(s) containing 'class GossipTopic' found, skipping changes for IGossipTopic
File(s) containing 'class GossipTopicCache' found, skipping changes for IGossipTopicCache
File(s) containing 'class Histogram' found, skipping changes for IHistogram
File(s) containing 'class HttpClient' found, skipping changes for IHttpClient
replacing IImportArgs
replacing IIndexAttestationData
replacing IInterchangeCompleteV4
replacing IInterchangeLodestar
replacing IInterchangeV5
replacing IJsonRpcBackend
File(s) containing 'class JsonRpcHttpClient' found, skipping changes for IJsonRpcHttpClient
replacing IKeyValue
replacing ILevelDbControllerMetrics
replacing ILevelDBOptions
replacing ILibp2pOptions
File(s) containing 'class LightClientStore' found, skipping changes for ILightClientStore
replacing ILogArgs
replacing ILogger
replacing ILoggerOptions
replacing IMarkdownSection
replacing IMetadataModules
replacing IMetadataOpts
replacing IMetricsArgs
File(s) containing 'class MinMaxSurround' found, skipping changes for IMinMaxSurround
replacing IMockChainParams
replacing InboundRateLimitQuota
File(s) containing 'class Network' found, skipping changes for INetwork
replacing INetworkArgs
replacing INetworkInitModules
replacing INetworkModules
replacing INetworkOptions
replacing InstanceCodeOptions
File(s) containing 'class PeerRpcScoreStore' found, skipping changes for IPeerRpcScoreStore
replacing IPeerRpcScoreStoreModules
replacing IPeerSummary
File(s) containing 'class PersistedKeysBackend' found, skipping changes for IPersistedKeysBackend
File(s) containing 'class Proof' found, skipping changes for IProof
replacing IQueueMetrics
replacing IReadonlyEpochShuffling
File(s) containing 'class ReqRespBeaconNode' found, skipping changes for IReqRespBeaconNode
replacing IRequestErrorMetadata
replacing IRewardPenaltyItem
replacing IRpcPayload
replacing IRpcResponse
replacing IRpcResponseError
replacing ISignedBLStoExecutionChange
replacing ISignedVoluntaryExitData
replacing ISimpleCase
replacing ISimpleStruct
File(s) containing 'class SlashingProtection' found, skipping changes for ISlashingProtection
replacing ISlotRange
replacing ISpecTestOptions
File(s) containing 'class StateRegenerator' found, skipping changes for IStateRegenerator
replacing ISubnetsService
replacing ISyncArgs
replacing ISyncCommitteeJob
replacing ISyncModule
replacing ISyncModules
replacing ITestCase
replacing IUncompressState
replacing IUser
replacing IValidatorEvents
replacing IValidatorGeneratorOpts
replacing IValidatorMonitor
replacing IWinstonInfoArg
replacing IWinstonInfoArg
```

### CODE REVIEW

File: packages/db/src/controller/metrics.ts
Comment (nflaig) [inline packages/db/src/controller/metrics.ts:2]: There is an inconsistency how metric interfaces outside of the beacon-node are defined. `ICounter` is updated to `Counter` whereas others such as `IHistogram` is not because it is implemented by a class in the beacon-node but not the actual package where it is used.

All the metric types defined in [state-transition metrics](https://github.com/ChainSafe/lodestar/blob/aa4349cee2b5bbefdf4e7c0bd58df36aaebff6de/packages/state-transition/src/metrics.ts) could remove the `I`-prefix. But maybe for consistency should always use interface with `I`-prefix for metrics

**Review by nflaig — DISMISSED** [review]
looks good to me, just some minor inconsistency in metric types we might wanna fix, see [comment](https://github.com/ChainSafe/lodestar/pull/5145#discussion_r1105426376)

nice script btw

**Comment by nflaig** [comment]
need to rebase against unstable and re-run script before merging

**Comment by dapplion** [comment]
Screenshot from 2023-02-20 16-17-08

I have manually double checked all files :rofl: all good but damn..

**Review by twoeths — DISMISSED** [review]

**Review by g11tech — DISMISSED** [review]

**Comment by nflaig** [comment]
same, was kinda painful 😂 but [load all file diffs](https://gist.github.com/juanca/5fd799c5b094e3e4f8b709cd101d7403) was a life saver

P.S: should not merge yet, there are few missing changes need to pull in from last commit of https://github.com/ChainSafe/lodestar/pull/5174

image

**Review by g11tech — APPROVED** [review]

**Review by twoeths — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.6.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.6.0) :tada:

**Pull request merged**
